core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -253,7 +253,7 @@ private DocumentMapper merge(DocumentMapper mapper, boolean updateAllTypes) {
                 // simulate first
                 MergeResult result = oldMapper.merge(mapper.mapping(), true, updateAllTypes);
                 if (result.hasConflicts()) {
-                    throw new MergeMappingException(result.buildConflicts());
+                    throw new IllegalArgumentException("Merge failed with failures {" + Arrays.toString(result.buildConflicts()) + "}");
```

Comment:
LoggerMessageFormat handles the toString on the array. It'd be ```throw new IllegalArgumentException(LoggerMessageFormat.format("Merge failed with failures {}", result.buildConflicts());```.

I you don't think it improves readability then I'm happy for you to merge as is.